File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.cpp
class Solution {
public:
    int minCost(string s, vector<int>& Time) {
        int ans = 0;
        for(int i = 1; s[i] != '\0'; i++){
            if(s[i - 1] == s[i]){
                ans += min(Time[i - 1], Time[i]);
                Time[i] = max(Time[i - 1],Time[i]);
            }
        }
        return ans;
    }
};

File: 1578-minimum-time-to-make-rope-colorful/1578-minimum-time-to-make-rope-colorful.py
class Solution(object):
    def minCost(self, colors, neededTime):
        ans, n,i = 0,len(colors),0
        while i < n:
            totSum = neededTime[i]
            maxi = neededTime[i]
            i += 1
            while i < n and colors[i - 1] == colors[i]:
                totSum += neededTime[i]
                maxi = max(maxi,neededTime[i])
                i += 1
            totSum -= maxi
            ans += totSum
        return ans



        """
        :type colors: str
        :type neededTime: List[int]
        :rtype: int
        """
        